util/dockerutil/docker.go

```diff
@@ -91,21 +92,54 @@ func DockerPullLocalImages(ctx context.Context, fe containerutil.ContainerFronte
 
 func dockerPullLocalImage(ctx context.Context, fe containerutil.ContainerFrontend, localRegistryAddr string, pullName string, finalName string) error {
 	fullPullName := fmt.Sprintf("%s/%s", localRegistryAddr, pullName)
+
 	err := fe.ImagePull(ctx, fullPullName)
 	if err != nil {
-		return errors.Wrapf(err, "image pull")
+		return errors.Wrap(err, "image pull")
+	}
+
+	// Fix for #2471 where Podman pulls seem exit before the image is available
+	// for tagging. Wait for the image to become available.
+	err = waitForImage(ctx, fe, fullPullName)
+	if err != nil {
+		return err
 	}
+
 	err = fe.ImageTag(ctx, containerutil.ImageTag{
 		SourceRef: fullPullName,
 		TargetRef: finalName,
 	})
 	if err != nil {
 		return errors.Wrap(err, "image tag after pull")
 	}
+
 	force := true // Sometimes Docker GCs images automatically (force prevents an error).
 	err = fe.ImageRemove(ctx, force, fullPullName)
 	if err != nil {
 		return errors.Wrap(err, "image rmi after pull and retag")
 	}
+
 	return nil
 }
+
+func waitForImage(ctx context.Context, fe containerutil.ContainerFrontend, fullName string) error {
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			m, err := fe.ImageInfo(ctx, fullName)
+			if err != nil {
+				// Not available. Retry.
+				time.Sleep(100 * time.Millisecond)
```

Comment:
Nit: Maybe do this?

```go
select {
        case <-ctx.Done():
        case <-time.After(100 * time.Millisecond):
}
```

Reply:
Oh.. I meant instead of the sleep. Such that if a cancellation takes place then, it would cancel quickly, instead of waiting for 100ms during the cancel.

Reply:
Gotcha. I just realized the latest change is basically the same as the ticker.